CONTRIBUTING.md

```diff
@@ -0,0 +1,228 @@
+You'll want to install Python 2.7.11, and the latest versions of Python 3.6 and 3.7
+
+Once pyenv and the three versions of Python are installed, install [tox](https://tox.readthedocs.io/en/latest/), our test runner.
+
+Finally, run tox with `tox`.  All tests should pass!
+
+
+## Contributing a new dataset loader
+
+To add a new dataset loader:
+1. Create an index in `mirdata/indexes`, e.g. `my_dataset_index.json`
+2. Create a module in `mirdata`, .e.g `my_dataset.py`
+
+### Creating an index
+
+The index should be a json file where the top level keys are the unique track
+ids of the datset. The values should be a dictionary of files associated with
+the track id, along with their checksums.
+
+Any file path included should be relative to the top level directory of the dataset.
+For example, if a dataset has the structure:
+```
+> Example_Dataset/
+    > audio/
+        track1.wav
+        track2.wav
+        track3.wav
+    > annotations/
+        track1.csv
+        Track2.csv
+        track3.csv
+```
+The top level directory is `Example_Dataset` and the relative path for `track1.wav`
+should be `Example_Dataset/audio/track1.wav`.
+
+Any unavailable field should be indicated with `null`.
+
+A possible index file for this example would be:
+```json
+{
+    "track1": {
+        "audio": [
+            "Example_Dataset/audio/track1.wav",
+            "912ec803b2ce49e4a541068d495ab570"
+        ],
+        "annotation": [
+            "Example_Dataset/annotations/track1.csv",
+            "2cf33591c3b28b382668952e236cccd5"
+        ]
+    },
+    "track2": {
+        "audio": [
+            "Example_Dataset/audio/track2.wav",
+            "65d671ec9787b32cfb7e33188be32ff7"
+        ],
+        "annotation": [
+            "Example_Dataset/annotations/Track2.csv",
+            "e1964798cfe86e914af895f8d0291812"
+        ]
+    },
+    "track3": {
+        "audio": [
+            "Example_Dataset/audio/track3.wav",
+            "60edeb51dc4041c47c031c4bfb456b76"
+        ],
+        "annotation": [
+            "Example_Dataset/annotations/track3.csv",
+            "06cb006cc7b61de6be6361ff904654b3"
+        ]
+  }
+}
+```
+
+In this example there is a (purposeful) mismatch between the name of the audio file
+`track2.wav` and its corresponding annotation file, `Track2.csv`, compared with
+the other pairs. *This mismatch should be included in the index*. This type of
+slight difference in filenames happens often in publicly available datasets, makes
+pairing audio and annotation files more difficult. We introduce use a fixed,
+version controlled index to account for this kind of mismatch, rather than relying
+on string parsing on load.
+
+Scripts used to create the dataset indexes are in the [scripts](https://github.com/mir-dataset-loaders/mirdata/tree/master/scripts) folder.
+
+
+### Creating a module.
+
+Copy and paste this template and fill it in for your dataset:
+
+```python
```

Comment:
Womp womp...this has changed... 

I'll help rewrite this when I finish my current PR

Reply:
🚀

Reply:
@rabitt, could I trouble you for some updated copy here?